src/main/java/org/dependencytrack/tasks/scanners/TrivyAnalysisTask.java

```diff
@@ -0,0 +1,553 @@
+            info.setPackageInfos(new PackageInfo[]{value});
+            if (os.get(key) != null) {
+                info.setOS(os.get(key));
+            }
+            infos.add(info);
+        });
+
+        CompletableFuture
+                    .runAsync(() -> analyzeBlob(map, infos.toArray(new BlobInfo[]{})), EXECUTOR)
+                    .whenComplete((result, exception) -> {
+                        countDownLatch.countDown();
+
+                        if (exception != null) {
+                            LOGGER.error("An unexpected error occurred while analyzing", exception);
+                        }
+                    });
+
+        try {
+            if (!countDownLatch.await(60, TimeUnit.MINUTES)) {
+                // Depending on the system load, it may take a while for the queued events
+                // to be processed. And depending on how large the projects are, it may take a
+                // while for the processing of the respective event to complete.
+                // It is unlikely though that either of these situations causes a block for
+                // over 60 minutes. If that happens, the system is under-resourced.
+                LOGGER.warn("The Analysis for project :" + components.get(0).getProject().getName() + "took longer than expected");
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean shouldAnalyze(final PackageURL packageUrl) {
+        return getApiBaseUrl()
+                .map(baseUrl -> !isCacheCurrent(Vulnerability.Source.NVD, apiBaseUrl, packageUrl.getCoordinates()))
+                .orElse(false);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void applyAnalysisFromCache(final Component component) {
+        getApiBaseUrl().ifPresent(baseUrl ->
+                applyAnalysisFromCache(Vulnerability.Source.NVD, apiBaseUrl,
+                        component.getPurl().getCoordinates(), component, getAnalyzerIdentity(), vulnerabilityAnalysisLevel));
+    }
+
+    private void analyzeBlob(final Map<String, Component> components, final BlobInfo[] blobs) {
+        for (final BlobInfo info : blobs) {
+
+            var blob = new PutRequest();
+            blob.setBlobInfo(info);
+
+            try {
+                MessageDigest digest = MessageDigest.getInstance("SHA-256");
+                //todo - need to establish standard and requirements from trivy
+                byte[] hash = digest.digest(java.util.UUID.randomUUID().toString().getBytes(StandardCharsets.UTF_8));
+
+                blob.setDiffID("sha256:" + Hex.encodeHexString(hash));
+                LOGGER.debug(blob.getDiffID());
+            } catch(NoSuchAlgorithmException e) {
+                LOGGER.error(e.getMessage());
+                blob.setDiffID("sha256:0000000000000000000000000000000000000000000000000000000000000000");
+            }
```

Comment:
Using `DigestUtils` lets you avoid having to handle the `NoSuchAlgorithm` exception, e.g.:

```suggestion
            String hash = DigestUtils.sha256Hex(java.util.UUID.randomUUID().toString());
            blob.setDiffID("sha256:" + hash);
```